lib/graphql/schema/input_object.rb

```diff
@@ -27,7 +27,28 @@ def initialize(values, context:, defaults_used:)
       attr_reader :arguments
 
       # Ruby-like hash behaviors, read-only
-      def_delegators :@ruby_style_hash, :to_h, :keys, :values, :each, :any?
+      def_delegators :@ruby_style_hash, :keys, :values, :each, :any?
+
+      def to_h
+        @ruby_style_hash.inject({}) do |h, (key, value)|
+          h.merge(key => unwrap_value(value))
```

Comment:
Should this do a `merge!` to avoid allocating a new hash for each entry? Same question in `unwrap_value`.